Add "kontena plugin upgrade" to upgrade all plugins

kke opened the pull request:
Fixes #2948 

Adds a `kontena plugin upgrade` command that can be used to upgrade plugins.

It has the option flag `--silent` for use in cronjobs.

kke commented:
Automatic updating would require some kind of "last_checked" timestamp to config and adding something like `Kontena.run('plugin update') if last_checked + interval > Time.now.to_i` to `bin/kontena`.

SpComb commented:
Isn't `kontena plugin update foo` basically equivalent to `kontena plugin install foo`? It's a little confusing having two ways to do the same thing, and also not that great in terms of code/test duplication.

The `kontena plugin update` functionality makes plenty of sense, need to test that. Although as an APT user, I would expect it to be `kontena plugin upgrade` ;)

kke commented:
Yep maybe it could be just for upgrading all.

kke commented:
ping @SpComb 

- Removed the option to `kontena plugin update xyz`, it always upgrades all. (use `kontena plugin install xyz` to upgrade individually)
- Renamed the sub-command from `kontena plugin update` to `kontena plugin upgrade`

SpComb commented on test/spec/features/plugin/upgrade_spec.rb:
```diff
@@ -0,0 +1,40 @@
+
+  it 'upgrades all plugins' do
+    k = run('kontena plugin upgrade')
+    expect(k.code).to be_zero
```
Consider `run!` for better error output on failures

SpComb replied:
Doesn't need this `expect(k.code).to be_zero` anymore

SpComb commented on test/spec/features/plugin/upgrade_spec.rb:
```diff
@@ -0,0 +1,40 @@
+
+  it 'does nothing if no updates' do
+    k = run('kontena plugin upgrade')
+    expect(k.code).to be_zero
```
Same here
